### ISSUE TITLE
Error while creating ApiDoc with api_products in payload through JSONAPI

### ISSUE THREAD

**Issue by shishir-intelli** [apigee/apigee-api-catalog-drupal#221]
## Description
Adding `api_products` to API doc through JSONAPI throwing error
`No ID specified for related resource`

## Steps to Reproduce
Create API doc by using Jsonapi, in request payload add api_products.

## Actual Behavior
`No ID specified for related resource`

## Expected Behavior
Add ApiDoc with api products successfully

## Notes
This issue is coming when using https://github.com/apigee/apigee-smartdocs-maven-plugin
Original Issue - https://github.com/apigee/apigee-smartdocs-maven-plugin/issues/55

**Comment by shishir-intelli** [comment]
@ssvaidyanathan 
We are working on new feature releases for ApigeeX and Drupal 10 releases for all the modules.
We have planned to work on this issue in Q3.

**Comment by ssvaidyanathan** [comment]
Can this be expedited as we have customers affected due to this issue

**Comment by shishir-intelli** [comment]
Certainly, We will prioritize this as soon as we release ApigeeX and Drupal10 new features.
